swift/extractor/main.cpp

```diff
@@ -51,7 +53,51 @@ static void lockOutputSwiftModuleTraps(const codeql::SwiftExtractorConfiguration
   }
 }
 
+// if `CODEQL_EXTRACTOR_SWIFT_RUN_UNDER` env variable is set, and either
+// * `CODEQL_EXTRACTOR_SWIFT_RUN_UNDER_FILTER` is not set, or
+// * it is set to a regexp matching any substring of the extractor call
+// then the running process is substituted with the command (and possibly
+// options) stated in `CODEQL_EXTRACTOR_SWIFT_RUN_UNDER`, followed by `argv`.
+// Before calling `exec`, `CODEQL_EXTRACTOR_SWIFT_RUN_UNDER` is unset to avoid
+// unpleasant loops.
+// An example usage is to run the extractor under `gdbserver :1234` when the
+// arguments match a given source file.
+void checkToRunUnderTool(int argc, char* const* argv) {
+  auto runUnder = getenv("CODEQL_EXTRACTOR_SWIFT_RUN_UNDER");
+  if (runUnder == nullptr) {
+    return;
+  }
+  auto runUnderFilter = getenv("CODEQL_EXTRACTOR_SWIFT_RUN_UNDER_FILTER");
+  if (runUnderFilter != nullptr) {
+    assert(argc > 0);
+    std::string call = argv[0];
+    for (auto i = 0; i < argc; ++i) {
```

Comment:
`argv[0]` gets appended twice at the beginning; `i` should start from 1.

Reply:
thanks for spotting this! I hesitated between two versions and ended up doing a wrong mix of the two :sweat_smile: